packages/lab/src/date-picker/useFocusOut.ts

```diff
@@ -0,0 +1,41 @@
+import type { ElementProps, FloatingContext } from "@floating-ui/react";
+import { useMemo } from "react";
+
+export interface UseKeyboardProps {
+  /**
+   * Whether the hook is enabled
+   * @default true
+   */
+  enabled?: boolean;
+}
+
+// Hook to close the floating element when focus is outside the reference and `floating` elements
```

Comment:
Is this the same as useDismiss from floating hi?

Reply:
No, as we need to keep it open, as long as any input, (single or range) or open calendar button have focus but close it when we tab out.